src/mount/mastercomm.cc

```diff
@@ -2118,6 +2121,13 @@ uint8_t fs_saureadchunk(std::vector<ChunkTypeWithAddress> &chunkservers, uint64_
 		setDisconnect(true);
 		return SAUNAFS_ERROR_IO;
 	}
+	for (auto& server : chunkservers) {
+		// If 127.0.0.1, let's assume it's the same as master
+		if (server.address.ip == localhost) {
+			server.address.ip=masterip;
+			safs::log_debug("changing chunkserver ip address 127.0.0.1 to {}", htonl(masterip));
+		}
+	}
```

Comment:
> and some test ?

I wasn't sure how to actually do this test, because our testing framework doesn't use local ip addresses (master never supported them until recently), but I'll try the test you send out. If it fails, I'll just write a unit test with some master mocking (but it might take a while).

> why MASTER is returning locahost on the FIRSTPLACE ?

Because master stores the chunkserver IP address used to connect and gives this directly to client, and that's it. Previously chunkserver failed to connect at all if it was on localhost, but that has changed.

Reply:
> to be honest I do not know what is a need/goal of your code change ....

Simple, allow client to connect to chunk servers that use the same localhost IP address as master.